File: Distribuido/APIRestVercel/model/Task.js
// model/Task.js
import pool from "../config/db.js";

export class Task {
  static async getAll() {
    const [rows] = await pool.query("SELECT * FROM tareas");
    return rows;
  }

  static async getById(id) {
    const [rows] = await pool.query("SELECT * FROM tareas WHERE id = ?", [id]);
    return rows[0];
  }

  static async create(task) {
    const { nombre, estatus, fecha_entrega } = task;
    const [result] = await pool.query(
      "INSERT INTO tareas (nombre, estatus, fecha_entrega) VALUES (?, ?, ?)",
      [nombre, estatus, fecha_entrega]
    );
    return { id: result.insertId, ...task };
  }

  static async update(id, task) {
    const { nombre, estatus, fecha_entrega } = task;
    await pool.query(
      "UPDATE tareas SET nombre = ?, estatus = ?, fecha_entrega = ? WHERE id = ?",
      [nombre, estatus, fecha_entrega, id]
    );
  }

  static async delete(id) {
    await pool.query("DELETE FROM tareas WHERE id = ?", [id]);
  }
}

File: Local/APIREST1/app/src/main/java/com/example/practica3room/model/Task.kt
package com.example.practica3room.model

import com.google.gson.annotations.SerializedName

data class Task(
        val id: Int? = null,

        @SerializedName("nombre")
        val nombre: String,

        @SerializedName("estatus")
        val estatus: Int,   // 0 o 1

        @SerializedName("fecha_entrega")
        val fechaEntrega: String   // "2025-11-20"
)
